module/src/main/scala/com/gu/googleauth/actions.scala

```diff
@@ -52,13 +50,54 @@ trait UserIdentifier {
     UserIdentity.fromRequest(request).filter(_.isValid || !authConfig.enforceValidity)
 }
 
-trait Actions extends UserIdentifier {
+object AuthAction {
+  type UserIdentityRequest[A] = AuthenticatedRequest[A, UserIdentity]
+}
+
+/**
+  * This action ensures that the user is authenticated and their token is valid. Is a user is not logged in or their
+  * token has expired then they will be authenticated.
+  *
+  * The AuthenticatedRequest will always have an identity.
+  *
+  * @param authConfig
+  * @param loginTarget The target that should be redirected to in order to carry out authentication
+  */
+class AuthAction[A](val authConfig: GoogleAuthConfig, loginTarget: Call)
+  (implicit bodyParser: BodyParser[A], val executionContext: ExecutionContext)
```

Comment:
My thinking was that `BodyParser` and `executionContext` are common enough that people might want to use implicits to resolve, but my preference is increasingly towards being explicit unless there's a significant benefit not to be. 

Not wedded to this decision though.